test/e2e/service_test.go

```diff
@@ -140,18 +140,22 @@ func serviceMultipleDelete(r *test.KnRunResultCollector, existService, nonexistS
 	assert.Check(r.T(), strings.Contains(out.Stdout, expectedErr), "Failed to get 'not found' error")
 }
 
-func serviceDeleteAll(r *test.KnRunResultCollector, service1 string, service2 string, service3 string) {
+func serviceDeleteAll(r *test.KnRunResultCollector) {
+	// Services to delete using kn service delete --all. Services created before using test.ServiceCreate.
+	services := []string{"svc1", "service2", "ksvc3"}
+
 	out := r.KnTest().Kn().Run("service", "list")
 	r.AssertNoError(out)
-	assert.Check(r.T(), strings.Contains(out.Stdout, service1), "The service ", service1, " does not exist (but is expected to exist)")
-	assert.Check(r.T(), strings.Contains(out.Stdout, service2), "The service ", service2, " does not exist (but is expected to exist)")
-	assert.Check(r.T(), strings.Contains(out.Stdout, service3), "The service ", service3, " does not exist (but is expected to exist)")
+	// Check if services created successfully by checking kn service list output.
+	for _, service := range services {
+		assert.Check(r.T(), strings.Contains(out.Stdout, service), "The service ", service, " does not exist (but is expected to exist)")
+	}
 
 	out = r.KnTest().Kn().Run("service", "delete", "--all")
 	r.AssertNoError(out)
-
 	namespace := r.KnTest().Kn().Namespace()
-	expectedSuccess := fmt.Sprintf("Service '%s' successfully deleted in namespace '%s'.\nService '%s' successfully deleted in namespace '%s'.\nService '%s' successfully deleted in namespace '%s'.\n",
-		service3, namespace, service2, namespace, service1, namespace)
-	assert.Check(r.T(), strings.Contains(out.Stdout, expectedSuccess), "Failed to get 'successfully deleted' message")
+	// Check if all services deleted successfully by checking kn service delete --all output.
+	for _, service := range services {
+		assert.Check(r.T(), strings.Contains(out.Stdout, "Service '"+service+"' successfully deleted in namespace '"+namespace+"'."), service+" not deleted from kn service delete --all (but is supposed to be)")
+	}
```

Comment:
IMO, we can frame this e2e into following 3 steps:
1. len(kn service list) > 0
2. kn service delete --all
3. len(kn service list) == 0

The output message checking could be as simple as checking `deleted` word.

Reply:
Thanks for the suggestions in making it simpler. Recent changes should follow the above flow, but let me know if there's any other feedback.